Speed up Grunt's file watching support

hectcastro opened the pull request:
This changeset overrides the default NFS `actimeo` to 2. According to the NFS manpage, `actimeo` is a shortcut setting for `acregmin`, `acregmax`, `acdirmin`, and `acdirmax`. All of those settings determine how quickly file and directory metadata are refreshed. Some of these default to as high as 30 or 60 seconds.

hectcastro committed:
Speed up Grunt's file watching support

This changeset overrides the default NFS `actimeo` to 2. According to
the NFS manpage, `actimeo` is a shortcut setting for `acregmin`,
`acregmax`, `acdirmin`, and `acdirmax`. All of those settings determine
how quickly file and directory metadata are refreshed. Some of these
default to as high as 30 or 60 seconds.

notthatbreezy commented:
+1 - @CloudNiner -- still having trouble provision with NFS?

CloudNiner commented:
Working on it, let you know in a few minutes.

CloudNiner commented:
Yes, I am. Weird.

hectcastro commented:
What's the error?

CloudNiner commented:
`mount.nfs: access denied by server while mounting 33.33.33.1`

flibbertigibbet commented:
Just noticed this PR was still hanging out here.  Should we go ahead and merge it?

hectcastro commented:
I think this is still open because we have yet to determine why setting the `actimeo` option prevents the NFS mount from working on @CloudNiner's workstation.

flibbertigibbet commented:
Oh, okay.  I thought perhaps he was having general problems with NFS that this was an attempt to address.

CloudNiner commented:
Looks like I might have a problem with nfsv4. Here are the options that work for me:

```
vers=3,noatime,actimeo=1
```

Here are the build/file sync times for a few combinations:

```
Host: instantaneous pickup of changes, ~8s sass build
VM, nfsv3, noatime, actimeo=1: instantaneous pickup of changes, ~16s sass build
VM, nfsv4: ~10s pickup of changes, ~12s sass build
```

I'd say we set the noatime,actimeo=1 flags, and then have an ENV var to optionally enable nfsv3?

Commit:
Support NFS version 3 and 4

This changeset adds support for NFS v3 via the
`NIH_WAYFINDING_NFS_VERSION` environment variable. If set to `3`, the
NFS mount will occur with NFS v3 support. In addition, the `noatime`
option was added and `actimeo` set to `1`.

hectcastro commented:
@CloudNiner Can you give it a spin with `NIH_WAYFINDING_NFS_VERSION=3`

CloudNiner commented:
+1 seems happy

CloudNiner commented:
Do we have a baseline Vagrantfile somewhere? This nfs config block should probably be added to all future projects that use NFS.

hectcastro commented:
Going to hold out for a +1 from someone who had this working on Linux to make sure it doesn't introduce a regression.

(I tested both paths locally, but I'm on a Mac.)
